components/ILIAS/LegalDocuments/classes/Administration.php

```diff
@@ -125,7 +133,46 @@ public function retrieveDocuments(): array
 
     public function retrieveIds(): array
     {
-        return $this->container->http()->wrapper()->post()->retrieve('ids', $this->container->refinery()->to()->listOf($this->container->refinery()->kindlyTo()->int()));
+        $ids = $this->container->http()->wrapper()->post()->retrieve(
+            'ids',
+            $this->container->refinery()->byTrying([
+                $this->container->refinery()->to()->listOf(
+                    $this->container->refinery()->kindlyTo()->int()
+                ),
+                $this->container->refinery()->always(null)
+            ])
+        );
+
+        if (!$ids) {
+            //Try reading from UI-Table action
+            $ids = $this->http_wrapper->query()->retrieve(
+                'legal_document_id',
+                $this->refinery->byTrying([
+                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int()),
+                    $this->refinery->always([]),
+                ])
+            );
+        }
+
+        if (!$ids) {
+            //Try reading from UI-Table "apply to all objects"
+            $ids = $this->http_wrapper->query()->retrieve(
```

Comment:
Why is this received as an array and not a simple string? Is it not expected to be a string containing `ALL_OBJECTS` without anything else or can there be other stuff too which must be ignored?

Reply:
because the ks ui-table action always sends it as an array

Reply:
Ah ok, I was thinking of `http...?legal_document_id=a,b,c....` instead of `http...?legal_document_id[]=a&legal_document_id[]=b...`. Which is already converted to a real array. My mistake.